src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicValue.cs

```diff
@@ -1,35 +1,282 @@
+
+    public static explicit operator float(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<float>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to Float");
+
+    public static explicit operator float?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<float?>();
+
+    public static explicit operator string?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<string>();
+
+    public static explicit operator ushort(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ushort>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to UInt32");
+
+    public static explicit operator ushort?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ushort?>();
+
+    public static explicit operator uint(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<uint>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to UInt32");
+
+    public static explicit operator uint?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<uint?>();
+
+    public static explicit operator ulong(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ulong>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to UInt64");
+
+    public static explicit operator ulong?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ulong?>();
+
+    public static explicit operator byte[]?(JsonDynamicValue value)
+    {
+        if (value?.JsonValue.GetObjectValue() is string str)
+        {
+            return Convert.FromBase64String(str);
+        }
+
+        throw new InvalidCastException($"Cannot convert {value} to Byte array");
+    }
+
+    public static explicit operator TimeSpan(JsonDynamicValue value)
     {
-        get
+        if (value?.JsonValue?.GetObjectValue() is string str)
         {
-            if (!_hasValue)
+            return TimeSpan.Parse(str, CultureInfo.InvariantCulture);
+        }
+
+        throw new InvalidCastException($"Cannot convert {value} to TimeSpan");
+    }
+
+    public static explicit operator TimeSpan?(JsonDynamicValue value)
+    {
+        var str = value?.JsonValue?.GetObjectValue() as string;
+
+        return str is not null
+        ? TimeSpan.Parse(str, CultureInfo.InvariantCulture)
+        : null;
+    }
+
+    public static explicit operator Uri?(JsonDynamicValue value)
+        => new Uri(value?.JsonValue?.GetValue<string>() ?? string.Empty);
+
+    private sealed class JsonDynamicMetaObject : DynamicMetaObject
+    {
+        public JsonDynamicMetaObject(Expression expression, JsonDynamicValue value)
+            : base(expression, BindingRestrictions.Empty, value)
+        {
+        }
+
+        public override DynamicMetaObject BindConvert(ConvertBinder binder)
+        {
+            var targetType = binder.Type;
+
+            var castMethod = typeof(JsonDynamicValue).GetMethods(BindingFlags.Public | BindingFlags.Static)
```

Comment:
Does it mean that each "cast" will do some reflection and then Expression interpretation? These expressions should be cached.

Reply:
Yes, that is true for dynamic invocations anyway. I added a cache for the reflection information. But the dynamic performance is bad anyway, so I originally thought that would not be necessary.